app/jobs/set_rights_job.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+##
+# Job to update rights for objects
+class SetRightsJob < GenericJob
+  ##
+  # A job that allows a user to update the rights for a list of druids
+  # @param [Integer] bulk_action_id GlobalID for a BulkAction object
+  # @param [Hash] params additional parameters that an Argo job may need
+  def perform(bulk_action_id, params)
+    super
+
+    @new_rights = params[:set_rights][:rights]
+
+    with_bulk_action_log do |log|
+      log.puts("#{Time.current} Starting SetRightsJob for BulkAction #{bulk_action_id}")
+      raise StandardError, 'Must provide rights' if @new_rights.blank?
+
+      update_druid_count
+
+      pids.each do |druid|
+        log.puts("#{Time.current} #{self.class}: Attempting #{druid} (bulk_action.id=#{bulk_action_id})")
+        set_rights(druid, log)
+      end
+      log.puts("#{Time.current} Finished SetRightsJob for BulkAction #{bulk_action_id}")
+    end
+  end
+
+  private
+
+  def set_rights(druid, log)
+    object_client = Dor::Services::Client.object(druid)
+    cocina_object = object_client.find
+    return unless verify_access(cocina_object, log)
+
+    # use dor services client to update the access
+    begin
+      state_service = StateService.new(druid, version: cocina_object.version)
+      raise StandardError, 'Object cannot be modified in its current state.' unless state_service.allows_modification?
+
+      object_client.update(params: cocina_object.new(cocina_update_attributes))
+      Argo::Indexer.reindex_pid_remotely(cocina_object.externalIdentifier)
```

Comment:
Please remove this line, since this is an async job, updating right away is not important. The async job will take care of it.

Reply:
I think I removed this with my latest commit?

Reply:
yep